rad/src/lib.rs

```diff
@@ -105,7 +106,10 @@ pub fn run_aggregation_report(
     let context = &mut ReportContext::default();
     context.stage = Stage::Aggregation;
 
-    let radon_script = unpack_radon_script(aggregate.script.as_slice())?;
+    let filters = aggregate.filters.as_slice();
+    let reducer = aggregate.reducer as u8;
```

Comment:
I would change `create_radon_script` to accept a `u32`, and instead of casting to `u8` here, use `u8::try_from` inside `create_radon_script`.